syntax/rust.vim

```diff
@@ -154,11 +153,11 @@ syn region    rustDerive      start="derive(" end=")" contained contains=rustDer
 syn keyword   rustDeriveTrait contained Clone Hash RustcEncodable RustcDecodable Encodable Decodable PartialEq Eq PartialOrd Ord Rand Show Debug Default FromPrimitive Send Sync Copy
 
 " dyn keyword: It's only a keyword when used inside a type expression, so
-" we make effort here to highlight it only when Rust identifiers follow it 
+" we make effort here to highlight it only when Rust identifiers follow it
```

Comment:
This and the one below it are not pertinent to the matter. Ideally, all the crate stuff will be in one commit, and this other stuff will either not be done, or be in a separate commit. Practice good commit hygiene! It’s fun! Sometimes it’s even useful afterwards! ☺

Reply:
Oops, I didn't even notice this, the editor did it automatically. I'll separate it out.

Reply:
A few years ago I got tired of that sort of thing happening, and so I started always using `--patch` on my `git add` invocations when modifying existing files (when adding new files you can’t), and verifying that the diff is in fact what I want to commit. With aliases, this means I now type `g ap`.

The staging area can definitely be helpful if you deliberately make it a part of your process.

Reply:
Sorry, it was my editor that introduced those extraneous white spaces. I recently moved to vim-plug and forgot to re-enable `bronson/vim-trailing-whitespace`.